docs/dev/install-docker.rst

```diff
@@ -0,0 +1,91 @@
+Developer Setup with Docker
+===========================
+
+The following describes how to set up an instance of the site on your
+computer for development with Docker and docker-compose.
+
+   .. Warning::
+
+    These installation steps are made for development only. It is not
+    recommended to run Pontoon via Docker in production.
+
+   .. Note::
+
+    Installation via Docker is still a fresh feature. Problems might happen,
+    bugs might be met. If you have any troubles with it, please
+    `file a bug <https://bugzilla.mozilla.org/enter_bug.cgi?product=Webtools&component=Pontoon>`_
+    or `propose a patch <https://github.com/mozilla/pontoon>`_!
+
+Prerequisites
+-------------
+
+1. Install `Docker <https://docs.docker.com/engine/installation/>`_.
+
+2. Install `docker-compose <https://docs.docker.com/compose/install/>`_. You need
+   1.10 or higher.
+
+3. Install `make <https://www.gnu.org/software/make/>`_ using either your
+   system's package manager (Linux) or homebrew (OSX) or FIXME (Windows).
+
+Quickstart
+----------
+
+1. From the root of this repository, run::
+
+     $ make dockerbuild
+
+   That will build the containers required for development: base and
+   webapp.
+
+2. Then run the webapp::
+
+      $ make dockerrun
+
+   .. Note::
+
+        The first time you run this, the PostgreSQL container needs to do
+        some work before it becomes available to the webapp container. Hence,
```

Comment:
I'm not really happy with healthchecks in docker, but if we'd add one for the postgresql container, the webapp would only start once that healthcheck succeeds.

The caveat is that the healthchecks only run at one single frequency, and thus you're either pounding your container with checks, or you wait for a while. As the first healthcheck is done after one cycle period.